tsdb/db_test.go

```diff
@@ -155,6 +155,65 @@ func TestDataAvailableOnlyAfterCommit(t *testing.T) {
 	testutil.Equals(t, map[string][]tsdbutil.Sample{`{foo="bar"}`: {sample{t: 0, v: 0}}}, seriesSet)
 }
 
+// TestNoPanicAfterWALCorrutpion ensures that querying the db after a WAL corruption doesn't cause a panic.
+// https://github.com/prometheus/prometheus/issues/7548
+func TestNoPanicAfterWALCorrutpion(t *testing.T) {
+	db, closeFn := openTestDB(t, &Options{WALSegmentSize: 32 * 1024}, nil)
+	defer func() {
+		closeFn()
+	}()
```

Comment:
Not a blocker, but we can start using the `t.Cleanup` method now. So this entire thing would turn out to be just `t.Cleanup(closeFn)` (I guess)

Reply:
yep good suggestion, updated